pkg/util/startup/retry.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// Package startup provides functionality for retrying kv operations on server
+// startup.
+//
+// Server startup is now heavily dependent on kv for retrieving various
+// metadata. With introduction of circuit breakers, it is now possible for
```

Comment:
This isn't new with circuit breakers: you could always get AmbiguousResultError and you needed to be ready to handle them. But of course circuit breakers can make this failure mode very, very sticky and so it now presents a much larger problem.